<a href="https://colab.research.google.com/github/2k25cse2512541-wq/assignment1-flyrank-ml/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/2k25cse2512541-wq/assignment1-flyrank-ml/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*


#My lane = Lane 2: Refresh / Content Opportunity Scoring
#why?->
#search visibility decays naturally as cintent becomes stale or intent shifts  Editorial teams have limited time and cannot manually audit every page. Lane 2 allows us to build an automated, evidence-backed review queue that prioritizes the pages highest in search demand and decay risk, maximizing editorial efficiency and preserving search traffic.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

##decision
Determining which decaying or underperforming high-value pages should be prioritized for human review, content refresh, expansion, or pruning.

##The Action:
An editor inspects the prioritized candidate queue, reviews its reason codes, and performs an update

##Cost of a Wrong Call:

Tagging a stable page: Editorial time and resources are wasted reviewing content that didn't need updating

Missing a real decline: High-traffic pages continue losing organic reach and revenue unnoticed.



In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd


url = "https://raw.githubusercontent.com/2k25cse2512541-wq/assignment1-flyrank-ml/main/data/raw/content_refresh_anonymized.csv"
df = pd.read_csv(url)

total_pages = len(df)
declining_pages = len(df[df["trend_direction"] == "down"])
pct_declining = (declining_pages / total_pages) * 100

total_impressions = df["impressions_90d"].sum()
declining_impressions = df[df["trend_direction"] == "down"]["impressions_90d"].sum()
pct_impressions_at_risk = (declining_impressions / total_impressions) * 100

stale_visible = len(df[(df["content_age_days"] >= 180) & (df["impressions_90d"] >= 500)])

print(f"1. Total Pages: {total_pages:,} | Declining Pages: {declining_pages:,} ({pct_declining:.1f}%)")
print(f"2. Total 90d Impressions at Risk: {declining_impressions:,} out of {total_impressions:,} ({pct_impressions_at_risk:.1f}%)")
print(f"3. High-Demand Stale Pages (Age >= 180d, Imps >= 500): {stale_visible:,}")

1. Total Pages: 30,000 | Declining Pages: 16,262 (54.2%)
2. Total 90d Impressions at Risk: 79,994,363 out of 156,010,989 (51.3%)
3. High-Demand Stale Pages (Age >= 180d, Imps >= 500): 9,929


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

##What I CAN claim :

Observed Traffic & Demand Decay: From the starter dataset ($30,000$ pages), we observed that $54.2\%$ ($16,262$ pages) are currently categorized with a declining trend direction.

Quantifiable Search Risk: Over $79.99\text{M}$ impressions ($51.3\%$ of total $90$-day search impressions) are associated with declining content, demonstrating a significant volume of search demand at risk.

Targeted Opportunity Pool: There are $9,929$ high-demand, stale pages (age $\ge 180$ days with $\ge 500$ impressions) that represent primary candidates for editorial audit.

Decision-Support Prioritization: The output serves as a prioritized review queue to help editorial and SEO teams allocate limited hours toward high-impact candidate pages.

##What I CANNOT claim:
No Causal Guarantee: We cannot claim that updating or refreshing a page will guarantee a recovery in search positions or traffic, as this analysis is purely observational and lacks an experimental A/B test or causal design.



Rule-Out Alternatives: We cannot claim every drop in traffic is a content quality issue without further checks for consolidation, seasonality, or SERP feature layout changes.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.